### PR TITLE
chore: add in filter for ec2 describe instances to include shutting down

### PR DESCRIPTION
**Pull request by njtran**
Fixes # 

**Description**
This updates karpenter-core to include https://github.com/aws/karpenter-core/pull/252 and is the buddy change.

The [AWS node lifecycle controller](https://github.com/kubernetes/cloud-provider-aws/blob/master/pkg/providers/v2/instances.go#L195-L197) only treats terminated instances as not found. Karpenter currently incorrectly considers nodes that are `ShuttingDown` as not found.  

Adding in the filter here means that our DescribeInstances call will now return instances that are shutting down. Since [we'll be terminating nodes that have a NotFound Instance](https://github.com/aws/karpenter-core/pull/252/files), we need to consider `ShuttingDown` instances as existing to let the pods/nodes for that instance to gracefully shut down.

This also adds in the check to return early if the underlying instance is not found for IsMachineDrifted()

**How was this change tested?**

* `make presubmit` 
* `make apply` && `make e2etests` for the buddy change

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [x] No

**Release Note**

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ellistarn** [comment]
Is this a chore or fix?

**Review by ellistarn — APPROVED** [review]

File: pkg/cloudprovider/cloudprovider.go [hunk 1]
```diff
@@ -228,7 +228,7 @@ func (c *CloudProvider) isAMIDrifted(ctx context.Context, machine *v1alpha5.Mach
 		return false, err
 	}
 	instance, err := c.instanceProvider.Get(ctx, instanceID)
-	if err != nil {
+	if cloudprovider.IgnoreMachineNotFoundError(err) != nil {
```
Comment (ellistarn) [inline pkg/cloudprovider/cloudprovider.go:231]: Won't this cause a problem on line 234, since we'll try to dereference *instance.ImageId

**Review by ellistarn — CHANGES_REQUESTED** [review]
Can you include some tests?

File: pkg/cloudprovider/cloudprovider.go [hunk 2]
```diff
@@ -193,7 +193,7 @@ func (c *CloudProvider) IsMachineDrifted(ctx context.Context, machine *v1alpha5.
 	}
 	amiDrifted, err := c.isAMIDrifted(ctx, machine, provisioner, nodeTemplate)
 	if err != nil {
-		return false, err
+		return false, cloudprovider.IgnoreMachineNotFoundError(fmt.Errorf("calculating ami drift, %w", err))
```
Comment (ellistarn) [inline pkg/cloudprovider/cloudprovider.go:196]: Does this work? This will ways be an error, since you're using fmt.Errorf.

Comment (njtran) [reply pkg/cloudprovider/cloudprovider.go:196]: Yes it does, wrapping works!

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Review by ellistarn — DISMISSED** [review]

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Review by ellistarn — DISMISSED** [review]

**Review by njtran — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Pull request merged**